test/ng/httpSpec.js

```diff
@@ -2364,17 +2364,45 @@ describe('$http param serializers', function() {
       expect(decodeURIComponent(jqrSer({a: 'b', foo: ['bar', 'baz']}))).toEqual('a=b&foo[]=bar&foo[]=baz');
     });
 
+    it('should serialize arrays with functions', function() {
+      expect(jqrSer({foo:[function() { return 'bar'; }]})).toEqual('foo%5B%5D=bar');
+    });
+
     it('should serialize objects by repeating param name with [key] suffix', function() {
       expect(jqrSer({a: 'b', foo: {'bar': 'barv', 'baz': 'bazv'}})).toEqual('a=b&foo%5Bbar%5D=barv&foo%5Bbaz%5D=bazv');
                                                                            //a=b&foo[bar]=barv&foo[baz]=bazv
     });
 
+    it('should serialize objects with function properties by repeating param name with [key] suffix', function() {
+      expect(jqrSer({a: function() { return 'b'; }})).toEqual(
+        'a=b');
+        //a=b
+    });
+
+    it('should serialize objects with function properties returning an object by repeating param name with [key] suffix', function() {
+      expect(jqrSer({a: function() { return { b: 'c' }; }})).toEqual(
+        'a=%7B%22b%22:%22c%22%7D');
```

Comment:
This is not exactly how jQuery does it, but that is not related to this PR :grin: